rtos/EventFlags.h

```diff
@@ -0,0 +1,89 @@
+ * Copyright (c) 2006-2012 ARM Limited
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+#ifndef EVENT_FLAGS_H
+#define EVENT_FLAGS_H
+
+#include <stdint.h>
+#include "cmsis_os2.h"
+#include "mbed_rtos1_types.h"
+#include "mbed_rtos_storage.h"
+
+namespace rtos {
+/** \addtogroup rtos */
+/** @{*/
+
+/** The EventFlags class is used to signal to whom it may concern about an event has occured.
+
+ @note
+ Memory considerations: The EventFlags control structures will be created on current thread's stack, both for the mbed OS
+ and underlying RTOS objects (static or dynamic RTOS memory pools are not being used).
+*/
+class EventFlags {
+public:
+    /** Create and Initialize a EventFlags object */
+    EventFlags();
+
+    /** Create and Initialize a EventFlags object
+
+     @param name name to be used for this EventFlags. It has to stay allocated for the lifetime of the thread.
+    */
+    EventFlags(const char *name);
+
+    /** Set the specified Event Flags.
+      @param   flags  specifies the flags that shall be set.
+      @return  event flags after setting or error code if highest bit set.
+     */
+    uint32_t set (uint32_t flags);
+
+    /** Clear the specified Event Flags.
+      @param   flags  specifies the flags that shall be cleared.
+      @return  event flags before clearing or error code if highest bit set.
+     */
+    uint32_t clear (uint32_t flags);
+
+    /** Get the current Event Flags.
+      @return  current event flags.
+     */
+    uint32_t get ();
+
+    /** Wait for one or more Event Flags to become signaled.
+      @param   flags    specifies the flags to wait for.
+      @param   options  specifies flags options (osFlagsXxxx).
+      @param   timeout  timeout value or 0 in case of no time-out.
+      @return  event flags before clearing or error code if highest bit set.
+     */
+    uint32_t wait (uint32_t flags, uint32_t options=osFlagsWaitAny, uint32_t timeout=0);
```

Comment:
I think following is reasonable:

```
#define flag1 0x1 
#define flag2 0x2

/*[...]*/
flag.wait(flag1) /* Waits only for the flag1 */
flag.wait(flag2) /* Waits only for the flag2 to be set */
res = flag.wait(0) /* Waits for anything to be set */
if (res | flag1 || res | flag2) { \*...*/ }
```

but we could also introduce `wait_any` or something similar to only interrupt the wait if any one of the flags finish. I think this form of API is easier and more user friendly.